site/pages/api/create-disruption.api.ts

```diff
@@ -1,119 +1,44 @@
-
-    const associatedLink = formFields["associated-link"];
-
-    if (associatedLink) {
-        validateAssociatedLink(associatedLink, errors, "some-error-id");
-    }
-
-    const disruptionReason: string = formFields["disruption-reason"] || "";
-
-    validateDisruptionReasons(disruptionReason, errors, "some-error-id");
-
-    const disruptionStartDate = formFields["disruption-start-date"];
-    const disruptionStartTime = formFields["disruption-start-time"];
-    const disruptionEndDate = formFields["disruption-end-date"];
-    const disruptionEndTime = formFields["disruption-end-time"];
-    const disruptionIsNoEndDateTime = formFields["disruption-no-end-date-time"];
-
-    validateDateTimeSection(
-        disruptionStartDate,
-        disruptionStartTime,
-        disruptionEndDate,
-        disruptionEndTime,
-        disruptionIsNoEndDateTime,
-        errors,
-        "some-error-id",
-    );
-    const publishStartDate = formFields["publish-start-date"];
-    const publishStartTime = formFields["publish-start-time"];
-    const publishEndDate = formFields["publish-end-date"];
-    const publishEndTime = formFields["publish-end-time"];
-    const publishIsNoEndDateTime = formFields["publish-no-end-date-time"];
-
-    validateDateTimeSection(
-        publishStartDate,
-        publishStartTime,
-        publishEndDate,
-        publishEndTime,
-        publishIsNoEndDateTime,
-        errors,
-        "some-error-id",
-    );
-
-    const disruptionData: DisruptionPageInputs = {
-        "type-of-disruption": disruptionType,
-        summary: summary,
-        description: description,
-        "associated-link": associatedLink,
-        "disruption-reason": disruptionReason,
-        "disruption-start-date": disruptionStartDate
-            ? getDateTime(disruptionStartDate).toString()
-            : disruptionStartDate,
-        "disruption-end-date": disruptionEndDate ? getDateTime(disruptionEndDate).toString() : disruptionEndDate,
-        "disruption-start-time": disruptionStartTime,
-        "disruption-end-time": disruptionEndTime,
-        "disruption-no-end-date-time": disruptionIsNoEndDateTime,
-        "publish-start-date": publishStartDate ? getDateTime(publishStartDate).toString() : publishStartDate,
-        "publish-end-date": publishEndDate ? getDateTime(publishEndDate).toString() : publishEndDate,
-        "publish-start-time": publishStartTime,
-        "publish-end-time": publishEndTime,
-        "publish-no-end-date-time": publishIsNoEndDateTime,
-        "disruption-repeats": formFields["disruption-repeats"],
-    };
-
-    setCookieOnResponseObject(COOKIES_DISRUPTION_INFO, JSON.stringify(disruptionData), res);
+    try {
+        const validatedBody = createDisruptionsSchemaRefined.parse(req.body);
+        setCookieOnResponseObject(COOKIES_DISRUPTION_INFO, JSON.stringify(validatedBody), res);
+        setCookieOnResponseObject(COOKIES_DISRUPTION_ERRORS, "", res, 0);
```

Comment:
Is this clearing a cookie? I wonder if it would be useful to have a function that just takes the cookie name as an argument, and it clears it. Just because we will be reusing this code a few places I think.